FRAuthenticator/FRAuthenticator/Manager/AuthenticatorManager.swift

```diff
@@ -46,6 +46,20 @@ struct AuthenticatorManager {
         let uriType = uri.getURIType()
         let authType = uri.getAuthType()
         if uriType == .mfauth {
+            FRALog.v("Validating stored Mechanisms for duplication")
+            do {
+                let parser = try PushQRCodeParser(url: uri)
+                let account = Account(issuer: parser.issuer, accountName: parser.label, imageUrl: parser.image, backgroundColor: parser.backgroundColor, policies: parser.policies)
+                for thisMechanism in self.storageClient.getMechanismsForAccount(account: account) {
+                    FRALog.e("Found a Mechanism under the same account")
+                    onError(MechanismError.alreadyExists(thisMechanism.identifier))
+                    return
```

Comment:
Even though this works, I'm not so sure about the use of for loop here. I saw in Android implementation we check the list for being empty and use the first item. Here also we could do something similar to improve the readability of the code. something like this: 
`if let thisMechanism = self.storageClient.getMechanismsForAccount(account: account).first { 
....log and process error and return
`
what do you think @rodrigoareis ?

Reply:
I agree.